extensions/wikia/AdEngine/js/AdInfoTracker.js

```diff
@@ -0,0 +1,135 @@
+/*global define*/
+define('ext.wikia.adEngine.adInfoTracker',  [
+	'ext.wikia.adEngine.adTracker',
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.lookup.services',
+	'ext.wikia.aRecoveryEngine.recovery.helper',
+	'wikia.log',
+	'wikia.window',
+	require.optional('ext.wikia.adEngine.mobile.mercuryListener')
+], function (adTracker, adContext, lookupServices, recoveryHelper, log, win, mercuryListener) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.adInfoTracker',
+		enabledSlots = {
+			TOP_LEADERBOARD: true,
+			HOME_TOP_LEADERBOARD: true,
+			TOP_RIGHT_BOXAD: true,
+			HOME_TOP_RIGHT_BOXAD: true,
+			PREFOOTER_LEFT_BOXAD: true,
+			PREFOOTER_MIDDLE_BOXAD: true,
+			PREFOOTER_RIGHT_BOXAD: true,
+			LEFT_SKYSCRAPER_2: true,
+			LEFT_SKYSCRAPER_3: true,
+			INCONTENT_BOXAD_1: true,
+			BOTTOM_LEADERBOARD: true,
+			MOBILE_TOP_LEADERBOARD: true,
+			MOBILE_BOTTOM_LEADERBOARD: true,
+			MOBILE_IN_CONTENT: true,
+			MOBILE_PREFOOTER: true
+		};
+
+	function shouldHandleSlot(slot) {
+		if (!enabledSlots[slot.name] ||
+			!slot.container.firstChild ||
+			!slot.container.firstChild.dataset.gptPageParams) {
+			return false;
+		}
+		return true;
+	}
+
+	function prepareData(slot, status) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 24 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)